src/Altinn.App.Core/Helpers/UserHelper.cs

```diff
@@ -50,28 +49,33 @@ public UserHelper(
     public async Task<UserContext> GetUserContext(HttpContext context)
     {
         using var activity = _telemetry?.StartGetUserContextActivity();
+        string? partyCookieValue = context.Request.Cookies[_settings.GetAltinnPartyCookieName];
+        Dictionary<string, string> tokenClaims = context.User.Claims.ToDictionary(
+            x => x.Type,
+            y => y.Value,
+            StringComparer.Ordinal
+        );
 
-        UserContext userContext = new UserContext() { User = context.User };
-
-        foreach (Claim claim in context.User.Claims)
+        UserContext userContext = new()
         {
-            if (claim.Type.Equals(AltinnCoreClaimTypes.UserName, StringComparison.Ordinal))
-            {
-                userContext.UserName = claim.Value;
-            }
-            else if (claim.Type.Equals(AltinnCoreClaimTypes.UserId, StringComparison.Ordinal))
+            User = context.User,
+            UserName = tokenClaims[AltinnCoreClaimTypes.UserName],
```

Comment:
this will throw `KeyNotFoundException` if the key doesn't exist, there is `GetValueOrDefault` which does what you want I think

Reply:
Gah, of course. Good eyes 👀